File: common/src/main/java/de/hype/bingonet/shared/constants/ChChestItem.java
package de.hype.bingonet.shared.constants;

/**
 * chchest Items. used to create custom ones which aren't in the {@link ChChestItems default list}
 */
public class ChChestItem {
    private String displayName;
    private String displayPath;
    private boolean custom;

    public ChChestItem(String displayName, String displayPath) {
        this.displayName = displayName;
        this.displayPath = displayPath;
        this.custom = false;
    }

    public ChChestItem(String displayName, boolean custom) {
        this.displayName = displayName;
        this.custom = custom;
        displayPath = null;
    }

    public String getDisplayName() {
        return displayName;
    }

    public ChChestItem setDisplayName(String displayName) {
        this.displayName = displayName;
        return this;
    }

    public boolean isCustom() {
        return custom;
    }

    public String getDisplayPath() {
        return displayPath;
    }

    @Override
    public String toString() {
        return displayName;
    }

    public boolean isGemstone() {
        return displayName.startsWith("Flawless") && displayName.endsWith("Gemstone");
    }

    public boolean isRoboPart() {
        String[] roboParts = {"Control Switch", "Electron Transmitter", "FTX 3070", "Robotron Reflector", "Superlite Motor", "Synthetic Heart"};
        for (String roboPart : roboParts) {
            if (displayName.equals(roboPart)) return true;
        }
        return false;
    }

    public boolean hasDisplayPath() {
        return displayPath != null;
    }

    public boolean isPowder() {
        return displayName.matches(".*Powder");
    }
}

File: common/src/main/java/de/hype/bingonet/shared/constants/ChChestItem.kt
package de.hype.bingonet.shared.constants

/**
 * chchest Items. used to create custom ones which aren't in the [default list][ChChestItems]
 */
class ChChestItem {
    var displayName: String
        private set
    val displayPath: String?
    val isCustom: Boolean

    constructor(displayName: String, displayPath: String?) {
        this.displayName = displayName
        this.displayPath = displayPath
        this.isCustom = false
    }

    constructor(displayName: String, custom: Boolean) {
        this.displayName = displayName
        this.isCustom = custom
        displayPath = null
    }

    fun setDisplayName(displayName: String): ChChestItem {
        this.displayName = displayName
        return this
    }

    override fun toString(): String {
        return displayName
    }

    val isGemstone: Boolean
        get() = displayName.startsWith("Flawless") && displayName.endsWith("Gemstone")

    val isRoboPart: Boolean
        get() {
            val roboParts = arrayOf<String?>(
                "Control Switch",
                "Electron Transmitter",
                "FTX 3070",
                "Robotron Reflector",
                "Superlite Motor",
                "Synthetic Heart"
            )
            for (roboPart in roboParts) {
                if (displayName == roboPart) return true
            }
            return false
        }

    fun hasDisplayPath(): Boolean {
        return displayPath != null
    }

    val isPowder: Boolean
        get() = displayName.matches(".*Powder".toRegex())
}